grass7/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -328,7 +339,18 @@ def download(self, output):
                 os.path.join(output, self._products_df_sorted['identifier'][idx])
             ))
             # download
-            self._api.download(self._products_df_sorted['uuid'][idx], output)
+            out = self._api.download(self._products_df_sorted['uuid'][idx],
+                                     output)
+            if sleep:
+                x = 1
+                online = out['Online']
+                while not online:
+                    time.sleep(int(sleep))
+                    out = self._api.download(self._products_df_sorted['uuid'][idx],
+                                             output)
+                    x += 1
+                    if x > maxretry:
+                        online = True
```

Comment:
This would mean that it is assumed that the product is online, even if it is not, forcing the process to fail, correct?
It would maybe be helpful to have a dedicated error message for this scenario, e.g. "Could not retrieve scene X after sleep time Y seconds and Z retries"

Reply:
Not to fail but to jump to the next scene, otherwise it will be in an infinite loop if the scene is not coming online (and it happen really often)